src/test/java/org/seqdoop/hadoop_bam/TestAnySAMInputFormat.java

```diff
@@ -0,0 +1,33 @@
+package org.seqdoop.hadoop_bam;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathNotFoundException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class TestAnySAMInputFormat {
+
+    @Test
+    public void testHeaderlessSamFormat() throws PathNotFoundException {
+        final SAMFormat result = getSamFormat(new Configuration(), "test_headerless.sam");
+        assertEquals(result, SAMFormat.SAM);
```

Comment:
Flip the arguments here, since for JUnit (unlike TestNG) the first argument should be the expected value, and the second the actual one.